packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,840 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+
+import "./interfaces/IElection.sol";
+import "../common/Initializable.sol";
+import "../common/FixidityLib.sol";
+import "../common/linkedlists/AddressSortedLinkedList.sol";
+import "../common/UsingRegistry.sol";
+
+
+contract Election is IElection, Ownable, ReentrancyGuard, Initializable, UsingRegistry {
+
+  using AddressSortedLinkedList for SortedLinkedList.List;
+  using FixidityLib for FixidityLib.Fraction;
+  using SafeMath for uint256;
+
+  struct TimestampedVote {
```

Comment:
What about "PendingVote"? it's not timestamped now.

Also, what if instead of saving the block where it was cast, we save the block where it can become active.